src/diagnostics/core_diagnostics.jl

```diff
@@ -733,3 +733,372 @@ add_diagnostic_variable!(
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* (cache.precomputed.cloud_diagnostics_tuple.q_ice),
+        )
+        return out
+    else
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* (cache.precomputed.cloud_diagnostics_tuple.q_ice),
+        )
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "clivi",
+    long_name = "Ice Water Path",
+    standard_name = "atmosphere_mass_content_of_cloud_ice",
+    units = "kg m-2",
+    comments = """
+    The total mass of ice in cloud per unit area.
+    (not just the area of the cloudy portion of the column). It doesn't include precipitating hydrometeors.
+    """,
+    compute! = compute_clivi!,
+)
+
+
+###
+# Vertical integrated dry static energy (2d)
+###
+function compute_dsevi!(out, state, cache, time)
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        cp = CAP.cp_d(cache.params)
+        z = Fields.coordinate_field(state.c.ρ).z
+        g = CAP.grav(cache.params)
+        dse = cache.scratch.ᶜtemp_scalar
+        @. dse =
+            state.c.ρ * (
+                cp * TD.air_temperature.(thermo_params, cache.precomputed.ᶜts) +
+                cache.core.ᶜΦ
+            )
+        Operators.column_integral_definite!(out, dse)
+        return out
+    else
+        cp = CAP.cp_d(cache.params)
+        z = Fields.coordinate_field(state.c.ρ).z
+        g = CAP.grav(cache.params)
+        dse = cache.scratch.ᶜtemp_scalar
+        @. dse =
+            state.c.ρ * (
+                cp * TD.air_temperature.(thermo_params, cache.precomputed.ᶜts) +
+                cache.core.ᶜΦ
+            )
+        Operators.column_integral_definite!(out, dse)
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "dsevi",
+    long_name = "Dry Static Energy Vertical Integral",
+    units = "",
+    comments = """
+    """,
+    compute! = compute_dsevi!,
+)
+
+###
+# column integrated cloud fraction (2d)
+###
+compute_clvi!(out, state, cache, time) =
+    compute_clvi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_clvi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("clvi", model)
+
+function compute_clvi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        cloud_cover = cache.scratch.ᶜtemp_scalar
+        cloud_cover .=
+            ifelse.(
+                cache.precomputed.cloud_diagnostics_tuple.cf .> 0.0f0,
+                1.0f0,
+                0.0f0,
+            )
```

Comment:
You are hardcoding a floating point precision here.

```suggestion
            FT = Spaces.undertype(axes(cloud_cover))
            ifelse.(
                cache.precomputed.cloud_diagnostics_tuple.cf .> zero(FT),
                one(FT),
                zero(FT),
            )
```

Same below